apps/client/src/features/operator/status-bar/StatusBar.module.scss

```diff
@@ -105,5 +105,7 @@
 }
 
 .progressOverride {
-  border-radius: 0;
+  & :global(.multiprogress-bar__bg) {
+    border-radius: 0;
+  }
```

Comment:
I am not sure breaking the encapsulation of the CSS modules is a good pattern to fall into. is there another alternative for us here?

Reply:
Good question, I wasn't sure the best way to expose border radius now that it is further into the DOM. In my testing it seemed pretty unavoidable that it had to be a layer down or the corner issue would come back.

As far as leakage goes the scope is still technically maintained here as the above compiles to 
```scss
.progressOverride-abc123 .multiprogress-bar__bg {}
``` 
but it's definitely not ideal so here are a few alternatives, tell me if you like any better or if anyone has a better idea.


1. Just make border radius a prop, I thought best to avoid that but could go that way if you want.

1. How about a css variable at the parent? Not sure if this is just more confusing to chase down, but could definititely go for something like...
    
  ```scss
  // StatusBar.module.scss
  .progressOverride {
    --border-radius: 0;
  }
  
  // MultiPartProgressBar.scss
  .multiprogress-bar__bg {
    border-radius: var(--border-radius, $progress-bar-br);
  }
  ```

Reply:
since the problem we are trying to solve is of styling, i believe the solution should not leak into the application code.
In that light, the CSS variable seems more indicated to me.

I am unsure on whether the common issues with CSS module loading order would end up being problematic for us, but assume that it may be a small edge case. Do you have experience with this?

On a side note, I am not actually sure what this code achieve, is it just a cleanup element for when the users have overridden the default colours?

We would ideally like to keep away from unnecessary complexity here. Do you think a solution that separates the elements as in the screenshot from JIRA in the ticket beneficial?
https://github.com/cpvalente/ontime/issues/900#issue-2246855661

Reply:
> since the problem we are trying to solve is of styling, i believe the solution should not leak into the application code. In that light, the CSS variable seems more indicated to me.

Sounds good!

> I am unsure on whether the common issues with CSS module loading order would end up being problematic for us, but assume that it may be a small edge case. Do you have experience with this?

For the variable method, my solution above would leave the css variable unset except when overridden like for the operator page, ie border radius would be set by the sass variable unless the css variable is defined. So I don't think module order should make a difference there as there will be no conflict and it should follow inheritance. I'll see if I can manage to test it.

> On a side note, I am not actually sure what this code achieve, is it just a cleanup element for when the users have overridden the default colours?

The CSS override code is just about matching the shading of the inactive part of the bar to the background behind, even when overridden. I didn't originally realize there were pages where the CSS override color was ignored, so that was the added fix. Even without the clipping issue, I would think it would be better for the bar to follow the background color this way. Would it make more sense to _opt in_ to the background override instead of opting out? (i.e. `allowCssOverrides`)

> We would ideally like to keep away from unnecessary complexity here. Do you think a solution that separates the elements as in the screenshot from JIRA in the ticket beneficial? [#900 (comment)](https://github.com/cpvalente/ontime/issues/900#issue-2246855661)

The Jira method is interesting, but I think it is more complex for this use case than needed because of the need for active time on top of the warning states. In order to avoid any div overlaps with this method, I think you'd have to further subdivide the sections (normal:past, normal:upcoming, warning:past, warning:upcoming, danger:past, danger:upcoming) and calculate the width for all 6 on each render. It is possible, but I think the current method is somewhat more flexible and future-proofed to allow for additional states or colors if needed. Does that sound sensible?

Reply:
Hi @hingobway , I apologise for the slow replies and hope this does not make it seem that I am not appreciative of your contribution.

From what I gather, your solution with CSS custom properties is solid and you have a good understanding of the reasonings and the technology.
I would suggest we follow with that solution 

> Would it make more sense to opt in to the background override instead of opting out? (i.e. allowCssOverrides)

Honestly, it likely doesnt matter. The component has limited use cases and I fail to see how the choice of one or the other has a meaningful impact on the ergonomics of using it. If I were to flip a coin, I would be inclined to say that options should modify behaviour (since CSSOverride is a natural behaviour, we would disable it instead)